app/routes/goal_routes.py

```diff
@@ -1 +1,119 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, abort, make_response, request, Response
+from ..models.goal import Goal
+from .route_utilities import validate_model, model_from_request
+from ..models.task import Task
+from ..db import db
+
+bp = Blueprint("goal_bp", __name__, url_prefix="/goals")
+
+
+@bp.post("")
+def create_goal():
+    new_goal = model_from_request(cls=Goal, request=request)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return {"goal": new_goal.to_dict()}, 201
+
+
+@bp.get("")
+def get_all_goals():
+    title_param = request.args.get("title")
+    sort = request.args.get("sort")
+
+    query = db.select(Goal)
+
+    if title_param:
+        query = query.where(goal.title.ilike(f"%{title_param}%"))
+
+    goals = db.session.scalars(query)
+
+    goals_response = []
+
+    for goal in goals:
+        goals_response.append(goal.to_dict())
+
+    return goals_response
```

Comment:
Could also use list comprehension here to make the code more concise.

```suggestion
    return [goal.to_dict() for goal in goals]
```